core/dutydb/memory.go

```diff
@@ -101,6 +109,24 @@ func (db *MemDB) Store(_ context.Context, duty core.Duty, unsignedSet core.Unsig
 		return errors.New("unsupported duty type", z.Str("type", duty.Type.String()))
 	}
 
+	// Delete all expired duties.
+	for {
+		var deleted bool
+		select {
+		case duty := <-db.deadliner.C():
+			err := db.deleteDutyUnsafe(duty)
+			if err != nil {
+				return err
+			}
```

Comment:
```suggestion
			err := db.deleteDutyUnsafe(duty); if err != nil {
				return err
			}
```

Reply:
not sure we have converged on a style for this pattern yet, we use both styles